Add a way to get unmapped reads from a BAM.

tomwhite opened the pull request:
Fixes #133.

heuermh commented on src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java:
```diff
@@ -72,31 +74,61 @@
 	// set this to true for debug output
 	public final static boolean DEBUG_BAM_SPLITTER = false;
 
+	static final String BOUNDED_TRAVERSAL_PROPERTY = "hadoopbam.bam.bounded-traversal";
```
Should this be `public static` as `INTERVALS_PROPERTY` is?  Also needs javadoc.

heuermh commented on src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java:
```diff
@@ -72,31 +74,61 @@
 	 */
 	public static final String INTERVALS_PROPERTY = "hadoopbam.bam.intervals";
 
+	static final String TRAVERSE_UNPLACED_UNMAPPED_PROPERTY = "hadoopbam.bam.traverse-unplaced-unmapped";
```
Should this be `public static` as `INTERVALS_PROPERTY` is?  Also needs javadoc.

heuermh commented on src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java:
```diff
@@ -72,31 +74,61 @@
+		setTraversalParameters(conf, intervals, false);
+	}
+
+	public static <T extends Locatable> void setTraversalParameters(Configuration conf,
```
Needs javadoc.

heuermh commented on src/test/java/org/seqdoop/hadoop_bam/TestBAMInputFormat.java:
```diff
@@ -26,11 +26,12 @@
   private JobContext jobContext;
   
-  private void completeSetup(List<Interval> intervals) {
+  private void completeSetup(boolean boundedTraversal, List<Interval> intervals, boolean
```
I know this is just test code, but with two boolean parameters it is not really clear in the test methods below which situation is under test.

How about adding new methods `setupBoundedTraversalUnplacedUnmapped(List<Interval>)` and similar that delegate to this one?

tomwhite commented:
@heuermh thanks for the review. I have addressed all your points: I added documentation to all methods, and also entries in the README to cover bounded traversal. I made the test code clearer, and also added a negative test for the case of mapped only reads (which are not currently supported).

Please let me know if it looks OK now.
